File: whisper/whisper.py
import os
import subprocess

# Define the source and output directories
source_dir = './source2Kgs'
output_dir = './output'

# Ensure the output directory exists
os.makedirs(output_dir, exist_ok=True)

# Iterate through all files in the source directory
for filename in os.listdir(source_dir):
    # Check if the file is an MP3 file
    if filename.endswith('.mp3'):
        input_file_path = os.path.join(source_dir, filename)
        
        # Construct the command to run
        command = [
            'whisper_timestamped', 
            input_file_path, 
            '--model', 'tiny', 
            '--language', 'ru', 
            '--output_dir', output_dir, 
            '--output_format', 'json'
        ]
        
        # Execute the command
        try:
            subprocess.run(command, check=True)
            print(f"Processed file: {filename}")
        except subprocess.CalledProcessError as e:
            print(f"Failed to process file {filename}: {e}")

File: whisper/whisper.js
const fs = require('fs');
const path = require('path');
const { exec } = require('child_process');

// Define the source and output directories
const sourceDir = path.join(__dirname, 'source2Kgs');
const outputDir = path.join(__dirname, 'output');

// Ensure the output directory exists
if (!fs.existsSync(outputDir)) {
  fs.mkdirSync(outputDir, { recursive: true });
}

// Get all files in the source directory
fs.readdir(sourceDir, (err, files) => {
  if (err) {
    console.error('Error reading source directory:', err);
    return;
  }

  // Filter for audio files (assuming .mp3 format)
  const audioFiles = files.filter(file => path.extname(file).toLowerCase() === '.mp3');

  // Process each audio file
  audioFiles.forEach(file => {
    const inputFilePath = path.join(sourceDir, file);
    const command = `whisper_timestamped "${inputFilePath}" --model tiny --language ru --output_dir "${outputDir}" --output_format "json"`;

    // Execute the command
    exec(command, (error, stdout, stderr) => {
      if (error) {
        console.error(`Error processing file ${file}:`, error);
        return;
      }
      if (stderr) {
        console.error(`stderr for file ${file}:`, stderr);
      }
      console.log(`Processed file ${file} successfully.`);
    });
  });
});
